central/productusage/service/service_impl.go

```diff
@@ -0,0 +1,83 @@
+package usage
+
+import (
+	"context"
+
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"github.com/pkg/errors"
+	datastore "github.com/stackrox/rox/central/productusage/datastore/securedunits"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/auth/permissions"
+	"github.com/stackrox/rox/pkg/grpc/authz"
+	"github.com/stackrox/rox/pkg/grpc/authz/perrpc"
+	"github.com/stackrox/rox/pkg/grpc/authz/user"
+	"github.com/stackrox/rox/pkg/sac/resources"
+	"google.golang.org/grpc"
+)
+
+var (
+	authorizer = perrpc.FromMap(map[authz.Authorizer][]string{
+		user.With(permissions.View(resources.Administration)): {
+			"/v1.ProductUsageService/GetCurrentSecuredUnitsUsage",
+			"/v1.ProductUsageService/GetMaxSecuredUnitsUsage",
+		}})
+)
+
+type serviceImpl struct {
+	v1.UnimplementedProductUsageServiceServer
+
+	datastore datastore.DataStore
+}
+
+// New returns a new Service instance using the given DataStore.
+func New(datastore datastore.DataStore) Service {
+	return &serviceImpl{
+		datastore: datastore,
+	}
+}
+
+// RegisterServiceServer registers this service with the given gRPC Server.
+func (s *serviceImpl) RegisterServiceServer(grpcServer *grpc.Server) {
+	v1.RegisterProductUsageServiceServer(grpcServer, s)
+}
+
+// RegisterServiceHandler registers this service with the given gRPC Gateway endpoint.
+func (s *serviceImpl) RegisterServiceHandler(ctx context.Context, mux *runtime.ServeMux, conn *grpc.ClientConn) error {
+	return errors.Wrap(v1.RegisterProductUsageServiceHandler(ctx, mux, conn), "failed to register the usage service handler")
+}
+
+// AuthFuncOverride specifies the auth criteria for this API.
+func (s *serviceImpl) AuthFuncOverride(ctx context.Context, fullMethodName string) (context.Context, error) {
+	return ctx, errors.Wrapf(authorizer.Authorized(ctx, fullMethodName), "failed to authorize a call to %s", fullMethodName)
```

Comment:
Just checking: is this error still transformed to 403/401?

Reply:
yes, 401:
```
HTTP/2 401 
content-type: application/json
vary: Accept-Encoding
content-length: 348
date: Fri, 18 Aug 2023 14:27:45 GMT

{"error":"failed to authorize a call to /v1.ProductUsageService/GetMaxSecuredUnitsUsage: credentials not found: not authorized: invalid username and/or password","code":16,"message":"failed to authorize a call to /v1.ProductUsageService/GetMaxSecuredUnitsUsage: credentials not found: not authorized: invalid username and/or password","details":[]}```